src/devices/machine/i82586.h

```diff
@@ -15,72 +15,377 @@ class i82586_base_device :
+
 	void port(u32 data); // cpu access interface
 
 protected:
-	i82596_base_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock, u8 datawidth, u8 addrwidth);
+	i82596_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock, endianness_t endian, u8 datawidth);
 
+	// standard device_* overrides
 	virtual void device_start() override;
 	virtual void device_reset() override;
 
+	// setup and scb
+	virtual void initialise() override;
+
+	// command unit
+	virtual bool cu_iasetup() override;
+	virtual bool cu_configure() override;
+	virtual bool cu_mcsetup() override;
+	virtual bool cu_transmit(u32 command) override;
+	virtual bool cu_tdreflect() override;
+	virtual bool cu_dump() override;
+
+	// receive unit
+	virtual bool address_filter(u8 *mac) override;
+	virtual void ru_execute(u8 *buf, int length) override;
+
+	// helpers
+	virtual u32 address(u32 base, int offset, int address) override { return (mode() == MODE_LINEAR) ? m_space->read_dword(base + address) : m_scb_base + m_space->read_word(base + offset); }
+	virtual u32 address(u32 base, int offset, int address, u16 empty) override;
+
 private:
+	// configure parameters
+	virtual u8 cfg_get(int offset) const override { return m_cfg_bytes[offset]; }
+	virtual void cfg_set(int offset, u8 data) override { m_cfg_bytes[offset] = data; }
+
+	bool cfg_crc_in_memory() const { return BIT(cfg_get(11), 2); }
+	bool cfg_mc_all()        const { return BIT(cfg_get(11), 5); }
+	bool cfg_multi_ia()      const { return BIT(cfg_get(13), 6); }
+
+	// initialisation, configuration and sysbus
+	inline sysbus_mode mode() const { return (sysbus_mode)(m_sysbus & SYSBUS_M); }
+
+	u8 m_cfg_bytes[CFG_SIZE];
+	u8 m_sysbus;
+	u32 m_iscp_address;
+
+	u64 m_mac_multi_ia; // multi-ia address hash table
 };
 
-class i82596sx_device : public i82596_base_device
+class i82596_le16_device : public i82596_device
 {
 public:
-	i82596sx_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
-
-protected:
-
-private:
+	i82596_le16_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
 };
 
-class i82596dx_device : public i82596_base_device
+class i82596_be16_device : public i82596_device
 {
 public:
-	i82596dx_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	i82596_be16_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+};
 
-protected:
+class i82596_le32_device : public i82596_device
+{
+public:
+	i82596_le32_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+};
 
-private:
+class i82596_be32_device : public i82596_device
+{
+public:
+	i82596_be32_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
 };
 
+ALLOW_SAVE_TYPE(i82586_base_device::cu_state);
+ALLOW_SAVE_TYPE(i82586_base_device::ru_state);
```

Comment:
Assuming these are for internal things, these should be in the source file, not the header.